commcare_connect/users/helpers.py

```diff
@@ -37,6 +38,13 @@ def create_hq_user(user, domain, api_key):
     )
     if hq_request.status_code == 201:
         return True
+    try:
+        hq_request.raise_for_status()
+    except httpx.HTTPStatusError as e:
+        raise CommCareHQAPIException(
+            f"{e.response.status_code} Error response {e.response.text} while creating user {user.username}"
+        )
+
     return False
```

Comment:
nit: can be refactored slightly:

```suggestion
    try:
        hq_request.raise_for_status()
    except httpx.HTTPStatusError as e:
        raise CommCareHQAPIException(
            f"{e.response.status_code} Error response {e.response.text} while creating user {user.username}"
        )

    return hq_request.status_code == 201
```